addons/l10n_no/models/account_invoice.py

```diff
@@ -31,10 +31,9 @@ def calculate_mod10(number):
             for idx, val in enumerate(number[::-1]):
                 for n in str(int(val) * [2, 1][idx % 2]):
                     total += int(n)
-            return number + str((10 - total % 10) % 10)
+            return number + str((10 - total) % 10)
 
         self.ensure_one()
-        partner_id = str(self.partner_id.id).zfill(7)
-        invoice_number = self.number.split('/')[-1].zfill(7)
-        ref = (partner_id + invoice_number)[:14]
-        return calculate_mod10(ref)
+        invoice_number = ''.join([i for i in self.number if i.isdigit()])
+        ref = (str(self.partner_id.id) + invoice_number[-7:]).zfill(14)
```

Comment:
What if the invoice_number is shorter than 7?   I prefer making sure partner and invoice number are 7 long first and concatenating those afterwards.

Reply:
sure, i'll do it.